packages/flutter_tools/lib/src/commands/daemon.dart

```diff
@@ -300,6 +301,48 @@ class DaemonDomain extends Domain {
   void dispose() {
     _subscription?.cancel();
   }
+
+  /// Enumerates the platforms supported by the provided project.
+  ///
+  /// This does not filter based on the current workflow restrictions, such
+  /// as whether command line tools are installed or whether the host platform
+  /// is correct.
+  Future<List<String>> getSupportedPlatforms(Map<String, dynamic> args) async {
+    final String projectRoot = _getStringArg(args, 'projectRoot', required: true);
+    final List<String> result = <String>[];
+    try {
+      final FlutterProject flutterProject = FlutterProject.fromDirectory(fs.directory(projectRoot));
+      if (flutterProject.linux.existsSync()) {
+        result.add('linux');
+      }
+      if (flutterProject.macos.existsSync()) {
```

Comment:
Should we also check the platform we're running on, so if someone commits a `macos` folder it doesn't cause it to show up as a target if you clone and try to run on Windows?

Reply:
In that case the project supports macos, but no device would support windows, so the filtering would happen on the device side.

Reply:
Ah yeah, that makes sense (and would match if we have metadata - you might load a project that claims it can target macOS but you're running on Windows) 👍